# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ooo/hotfix_xmlhelp-port-to-db-4.3.patch
# Copyright (C) 2004 - 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./patches/src680/xmlhelp-port-to-db-4.3.diff.orig	2005-10-31 11:23:06.000000000 -0300
+++ ./patches/src680/xmlhelp-port-to-db-4.3.diff	2006-01-06 14:43:18.000000000 -0300
@@ -335,27 +335,24 @@
  			System.out.println("Error initializing database");
  			System.exit(1);
  		} catch (FileNotFoundException fnfe) {
-@@ -429,9 +431,12 @@
- 				list[j++] = (String) enum.nextElement();
+@@ -405,10 +405,8 @@
+ 				list[j++] = (String) enumer.nextElement();
  			}
  
 -			Db table;
 +			Database table;
  			try {
 -				table = new Db(null, 0);
-+				DatabaseConfig cfg=DatabaseConfig.DEFAULT;
-+				cfg.setType(DatabaseType.BTREE);
-+				cfg.setAllowCreate(true);
-+				cfg.setMode(0644);
- 
+-
  				String fileName =
  					HelpDatabases.getInstallDirectory()
-@@ -440,15 +445,15 @@
+ 						+ _language
+@@ -416,15 +414,15 @@
  						+ _module
  						+ ".key";
  
 -				table.open(null,fileName, null, Db.DB_BTREE, Db.DB_CREATE, 0644);
-+				table = new Database(fileName, null, cfg);
++				table = new Database(fileName, null);
  
  				for (int i = 0; i < list.length; ++i) {
  					Data data = (Data) _hash.get(list[i]);
@@ -369,18 +366,21 @@
  			} catch (Exception e) {
  				System.out.println("error writing keydata");
  			}
-@@ -464,10 +469,8 @@
- 			list[j++] = (String) enum.nextElement();
- 		}
+@@ -429,9 +431,12 @@
+ 				list[j++] = (String) enum.nextElement();
+ 			}
  
--		Db table;
-+		Database table;
- 		try {
--			table = new Db(null, 0);
--
- 			String fileName =
- 				HelpDatabases.getInstallDirectory()
- 					+ _language
+-			Db table;
++			Database table;
+ 			try {
+-				table = new Db(null, 0);
++				DatabaseConfig cfg=DatabaseConfig.DEFAULT;
++				cfg.setType(DatabaseType.BTREE);
++				cfg.setAllowCreate(true);
++				cfg.setMode(0644);
+ 
+ 				String fileName =
+ 					HelpDatabases.getInstallDirectory()
 @@ -475,15 +478,19 @@
  					+ _module
  					+ ".ht";
